File: python/10001-20000/10828.py
stack = []
def empty(stack): return 0 if len(stack) else 1

for i in range(int(input())):
	command = input().split(" ")
	if len(command) != 2:
		if command[0] == "top":
			if not empty(stack): print(stack[len(stack)-1])
			else: print("-1")
		if command[0] == "size":
			print(len(stack))
		if command[0] == "empty":
			print(empty(stack))
		if command[0] == "pop":
			if not empty(stack):
				print(stack[len(stack)-1])
				del stack[len(stack)-1]
			else: print("-1")
	else: stack.append(command[1])

File: Solved/10001-20000/10828/10828.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main() {
	int T, X;
	string cmd;
	stack<int> s;
	
	cin >> T;
	while (T--) {
		cin >> cmd;
		
		if (cmd == "push") {
			cin >> X;
			s.push(X);
		}
		
		else if (cmd == "pop" || cmd == "top") {
			if (s.empty()) cout << "-1\n";
			else {
				cout << s.top() << "\n";
				if (cmd == "pop") s.pop();
			}
		}
		
		else if (cmd == "size") cout << s.size() << "\n";
		else if (cmd == "empty") cout << (s.empty() ? "1\n":"0\n");
	}

	return 0;
}